org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/backend/filebased/JsonBasedMultiNamespaceManager.java

```diff
@@ -0,0 +1,177 @@
+        }
+        INSTANCE = new JsonBasedMultiNamespaceManager(repository);
+        return INSTANCE;
+    }
+
+    private Map<FilebasedRepository, Map<String, NamespaceProperties>> loadNamespacePropertiesFromFile() {
+        Map<FilebasedRepository, Map<String, NamespaceProperties>> result = new HashMap<>();
+        Map<String, NamespaceProperties> nsProps = new HashMap<>();
+
+        for (FilebasedRepository repo : this.repository.getRepositories()) {
+            RepositoryFileReference ref = BackendUtils.getRefOfJsonConfiguration(new NamespacesId());
+            File file = repo.ref2AbsolutePath(ref).toFile();
+            try {
+                if (file.exists()) {
+                    TypeReference<HashMap<String, NamespaceProperties>> hashMapTypeReference =
+                        new TypeReference<HashMap<String, NamespaceProperties>>() {
+                        };
+                    nsProps = objectMapper.readValue(file, hashMapTypeReference);
+                }
+            } catch (IOException e) {
+                LOGGER.debug("Error while loading the namespace file.", e);
+                throw new RuntimeException();
+            }
+            result.put(repo, nsProps);
+        }
+
+        return result;
+    }
+
+    @Override
+    public void addAllPermanent(Collection<NamespaceProperties> properties) {
+        properties.forEach(prop -> {
+            FilebasedRepository repository = this.repository.getRepositoryUtils().getRepositoryByNamespace(prop.getNamespace());
+
+            if (prop.getUpstreamRepository().isEmpty() && repository instanceof GitBasedRepository) {
+                prop.setUpstreamRepository(((GitBasedRepository) repository).getRepositoryUrl());
+            }
+
+            repository.getNamespaceManager().addAllPermanent(Collections.singletonList(prop));
+        });
+        this.repository.updateNamespaces();
+    }
+
+    @Override
+    public void replaceAll(Map<String, NamespaceProperties> map) {
+        map.forEach((namespace, properties) -> {
+            FilebasedRepository repository = this.repository.getRepositoryUtils().getRepositoryByNamespace(namespace);
+            repository.getNamespaceManager().replaceAll(map);
+        });
+        this.repository.updateNamespaces();
+    }
+
+    @Override
+    public void clear() {
+        for (FilebasedRepository repo : this.repository.getRepositories()) {
+            System.out.println("\t\t\tCLEAR REPO NS --- " + repo.getRepositoryRoot());
+            repo.getNamespaceManager().clear();
+        }
+    }
+
+    @Override
+    public Map<String, NamespaceProperties> getAllNamespaces() {
+        Map<String, NamespaceProperties> result = new HashMap<>();
+        for (FilebasedRepository repo : this.repository.getRepositories()) {
+            result.putAll(repo.getNamespaceManager().getAllNamespaces());
+        }
+        return result;
+    }
+
+    @Override
+    protected Set<String> getAllPrefixes(String namespace) {
```

Comment:
you are not using the `namespace` argument in here. Could you add a comment explaining why?